Sitemap/ChainProvider.php

```diff
@@ -0,0 +1,67 @@
+<?php
+
+/*
+ * This file is part of the Symfony CMF package.
+ *
+ * (c) 2011-2014 Symfony CMF
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Cmf\Bundle\SeoBundle\Sitemap;
+
+use Symfony\Cmf\Bundle\SeoBundle\Model\UrlInformation;
+
+/**
+ * Accepts providers and merges the result of all providers into
+ * a combined list of UrlInformation
+ *
+ * @author Maximilian Berghoff <Maximilian.Berghoff@gmx.de>
+ */
+class ChainProvider implements UrlInformationProviderInterface
+{
+    /**
+     * @var UrlInformationProviderInterface[][]
+     */
+    private $providers = array();
+
+    /**
+     * {@inheritDoc}
+     */
+    public function addProvider(UrlInformationProviderInterface $provider, $priority = 0)
+    {
+        if (empty($this->providers[$priority])) {
+            $this->providers[$priority] = array();
+        }
+        $this->providers[$priority][] = $provider;
```

Comment:
$this->sortedProviders would be the cache for the providers, needs to be reset each time a provider is added when doing caching